mlrun/api/db/sqldb/db.py

```diff
@@ -2856,10 +3735,66 @@ def patch_feature_vector(
         )
 
     def delete_feature_vector(self, session, project, name, tag=None, uid=None):
-        self._delete_feature_store_object(
-            session, FeatureVector, project, name, tag, uid
+        self._delete_tagged_object(
+            session,
+            FeatureVector,
+            project=project,
+            tag=tag,
+            uid=uid,
+            name=name,
         )
 
+    def _delete_tagged_object(
+        self,
+        session,
+        cls,
+        project,
+        tag=None,
+        uid=None,
+        commit=True,
+        **kwargs,
+    ):
+        if tag and uid:
+            raise mlrun.errors.MLRunInvalidArgumentError(
+                "Both uid and tag specified when deleting an object."
+            )
+
+        object_id = None
+        obj_name = kwargs.get("name") or kwargs.get("key")
```

Comment:
It's not critical, it just seemed strange to me that we look for specific kwargs (which are usually just opaque to the code). If this will complicate the code then I'm ok with keeping this as-is, though I'm not sure where this adds complexity. If you just replace this line with 
`obj_name = name or key`
And pass `name=name, key=key` to the filter query, what other changes are needed?